### PR TITLE
Implement gdriveplayer and gomostream

### PR DESCRIPTION
**Pull request by JamesHawkinss**

### CODE REVIEW

File: src/providers/list/gdriveplayer/index.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    ).find((e) => e.textContent?.includes("eval"));
+
+    if (!script || !script.textContent) {
+      throw new Error("Could not find stream");
+    }
+
+    const data = unpack(script.textContent).split("var data=\\'")[1].split("\\'")[0].replace(/\\/g, "");
+    const decryptedData = unpack(CryptoJS.AES.decrypt(data, "alsfheafsjklNIWORNiolNIOWNKLNXakjsfwnBdwjbwfkjbJjkopfjweopjASoiwnrflakefneiofrt", { format }).toString(CryptoJS.enc.Utf8));
+    // eslint-disable-next-line
+    const sources = JSON.parse(JSON.stringify(eval(decryptedData.split("sources:")[1].split(",image")[0].replace(/\\/g, "").replace(/document\.referrer/g, "\"\""))));
```
Comment (mrjvs) [inline src/providers/list/gdriveplayer/index.ts:85]: Pain

File: src/providers/list/gomostream/index.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  async searchForMedia(query: MWQuery): Promise<MWProviderMediaResult[]> {
+    const term = query.searchQuery.toLowerCase();
+
+    const params = new URLSearchParams({
+      apikey: OMDB_API_KEY,
+      s: term,
+      type: query.type
+    });
+    const searchRes = await fetch(
+      `${CORS_PROXY_URL}http://www.omdbapi.com/?${encodeURIComponent(params.toString())}`,
```
Comment (mrjvs) [inline src/providers/list/gomostream/index.ts:47]: What if gomostream doesn't have the search results from the omdb api?

File: src/providers/list/gdriveplayer/index.ts (hunk 1 above)
Comment (JamesHawkinss) [reply src/providers/list/gdriveplayer/index.ts:85]: i either didn't write this or don't care enough to remember writing it

Comment (mrjvs) [reply src/providers/list/gdriveplayer/index.ts:85]: Add least add a comment or todo about needing to rewrite. This is dangerous, especially as its third part code, there could be a targeted attack against our users

File: src/providers/list/gomostream/index.ts (hunk 2 above)
Comment (JamesHawkinss) [reply src/providers/list/gomostream/index.ts:47]: there are many conditions that could cause it to not have the content, and if we check them all then we'll basically be duplicating the amount of api requests that we have to do... we might just have to let users click it and be disappointed if it doesn't work

gomostream can fail in like 5 different places, and can also fail when grabbing the final stream url. unreliable.

File: src/providers/types.ts [hunk 3]
```diff
@@ -20,7 +20,8 @@ export interface MWMediaCaption {
 }
 export interface MWMediaStream {
   url: string;
-  type: MWMediaStreamType;
+  // type: MWMediaStreamType;
+  type: string;
```
Comment (mrjvs) [inline src/providers/types.ts:24]: Don't remove my type ;-;
Throw error in scraper if unsupported type

File: src/providers/list/gomostream/index.ts (hunk 2 above)
Comment (mrjvs) [reply src/providers/list/gomostream/index.ts:47]: At a later date, I'd want to get rid of gomo. It seems like a terrible source

**Review by mrjvs — CHANGES_REQUESTED** [review]

File: src/providers/types.ts (hunk 3 above)
Comment (JamesHawkinss) [reply src/providers/types.ts:24]: will dm you

**Pull request merged**